lib/active_merchant/billing/gateways/decidir_plus.rb

```diff
@@ -0,0 +1,151 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class DecidirPlusGateway < Gateway
+      self.test_url = 'https://developers.decidir.com/api/v2'
+      self.live_url = 'https://live.decidir.com/api/v2'
+
+      self.supported_countries = ['ARG']
+      self.default_currency = 'ARS'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'http://decidir.com.ar/home'
+      self.display_name = 'Decidir Plus'
+
+      def initialize(options = {})
+        requires!(options, :public_key, :private_key)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        post = {}
+        post[:site_transaction_id] = options[:site_transaction_id]
```

Comment:
Regarding the site_transaction_id, since it is a required field, have we considered creating one via SecureRandom.hex if one is not provided from the merchant?

Reply:
Good catch, I think I left this out thinking we might pass in a GSF and handle the randomly generated value in Core, but the behavior makes it clear here what the gateway expects.